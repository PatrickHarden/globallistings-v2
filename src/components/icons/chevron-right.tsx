import React, { FC } from 'react';

interface Props {
  readonly fill?: string;
  readonly height?: string;
  readonly width?: string;
}

export const ChevronRight: FC<Props> = ({ fill = '#000', height = '21', width = '10' }) => (
  <svg
    width={width}
    height={height}
    viewBox={`0 0 ${width} ${height}`}
    version="1.1"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
      <g transform="translate(-221.000000, -315.000000)" fill={fill}>
        <g transform="translate(16.000000, 161.000000)">
          <g transform="translate(16.000000, 139.000000)">
            <g transform="translate(185.000000, 15.000000)">
              <path
                d="M5.92862876,7.95533112 L12.9953224,13.8849911 L11.9668622,15.1106622 L4,8.42567111 L4.00449273,8.42031688 L4,8.41904297 L4,7.48499114 L11.9668622,0.8 L12.9953224,2.02567111 L5.92862876,7.95533112 Z"
                transform="translate(8.497661, 7.955331) scale(-1, 1) translate(-8.497661, -7.955331) "
              />
            </g>
          </g>
        </g>
      </g>
    </g>
  </svg>
);
