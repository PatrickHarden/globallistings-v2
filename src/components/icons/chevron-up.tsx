import React, { FC } from 'react';

interface Props {
  readonly fill?: string;
  readonly height?: string;
  readonly width?: string;
}

export const ChevronUp: FC<Props> = ({ fill = '#006A4D', height = '10', width = '16' }) => (
  <svg
    width={width}
    height={height}
    viewBox={`0 0 ${width} ${height}`}
    version="1.1"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
      <g transform="translate(-98.000000, -16.000000)" fill={fill}>
        <g transform="translate(106.000000, 21.000000) rotate(90.000000) translate(-106.000000, -21.000000) translate(98.000000, 13.000000)">
          <path d="M5.12862876,7.95533112 L12.1953224,13.8849911 L11.1668622,15.1106622 L3.2,8.42567111 L3.20449273,8.42031688 L3.2,8.41904297 L3.2,7.48499114 L11.1668622,0.8 L12.1953224,2.02567111 L5.12862876,7.95533112 Z" />
        </g>
      </g>
    </g>
  </svg>
);
