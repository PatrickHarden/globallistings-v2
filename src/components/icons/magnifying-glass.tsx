import React, { FC } from 'react';

interface Props {
  readonly backgroundFill?: string;
  readonly fill?: string;
  readonly height?: string;
  readonly width?: string;
}

export const MagnifyingGlass: FC<Props> = ({ fill = '#006A4D', height = '24', width = '24' }) => (
  <svg
    width={width}
    height={height}
    viewBox={`0 0 ${width} ${height}`}
    version="1.1"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
      <g transform="translate(-52.000000, -244.000000)" fill={fill}>
        <g transform="translate(16.000000, 161.000000)">
          <g transform="translate(16.000000, 67.000000)">
            <g>
              <g transform="translate(16.000000, 12.000000)">
                <g>
                  <g transform="translate(4.000000, 4.000000)">
                    <path
                      d="M9,18 C4.02943725,18 0,13.9705627 0,9 C0,4.02943725 4.02943725,0 9,0 C13.9705627,0 18,4.02943725 18,9 C18,13.9705627 13.9705627,18 9,18 Z M9,16 C12.8659932,16 16,12.8659932 16,9 C16,5.13400675 12.8659932,2 9,2 C5.13400675,2 2,5.13400675 2,9 C2,12.8659932 5.13400675,16 9,16 Z"
                      fillRule="nonzero"
                    />
                    <rect
                      transform="translate(18.949747, 18.949747) rotate(-45.000000) translate(-18.949747, -18.949747) "
                      x="17.9497475"
                      y="12.9497475"
                      width="2"
                      height="12"
                    />
                  </g>
                </g>
              </g>
            </g>
          </g>
        </g>
      </g>
    </g>
  </svg>
);

/*


  <svg
    width={width}
    height={height}
    viewBox="0 0 17 16"
    version="1.1"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
      <g transform="translate(-38.000000, -17.000000)">
        <g transform="translate(38.000000, 17.000000)">
          <path
            stroke={fill}
            strokeWidth="3"
            strokeLinecap="round"
            d="M7.40625,8.140625 L15,14.46875"
          />
          <g>
            <path
              fill={backgroundFill}
              fillRule="evenodd"
              d="M14,7 C14,3.13400656 10.8659934,0 7,0 C3.13400659,0 0,3.13400656 0,7 C0,10.8659934 3.13400659,14 7,14 C10.8659934,14 14,10.8659934 14,7 Z"
            />
            <path
              stroke={fill}
              strokeWidth="3"
              d="M12.5,7 C12.5,3.96243369 10.0375663,1.5 7,1.5 C3.9624337,1.5 1.5,3.96243369 1.5,7 C1.5,10.0375663 3.96243371,12.5 7,12.5 C10.0375663,12.5 12.5,10.0375663 12.5,7 Z"
            />
          </g>
        </g>
      </g>
    </g>
  </svg>
  */
