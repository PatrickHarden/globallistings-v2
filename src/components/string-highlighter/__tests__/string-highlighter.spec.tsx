import React from 'react';
import { shallow } from 'enzyme';

import StringHighlighter from '../string-highlighter';
import { ArrowRight } from '../../icons/arrow-right';

describe('components', () => {
  describe('string-highlighter', () => {
    const props = {
      enabled: true,
      highlightSection: 'a str',
      str: 'This is a string',
    };

    it('should render correctly', () => {
      expect(shallow(<StringHighlighter {...props} />)).toMatchSnapshot();
    });

    it('should render correctly when there is no match', () => {
      expect(shallow(<StringHighlighter {...props} highlightSection="foo" />)).toMatchSnapshot();
    });

    it('should render correctly with a custom element', () => {
      expect(shallow(<StringHighlighter {...props} highlightComponent="span" />)).toMatchSnapshot();
    });

    it('should render correctly with a custom class', () => {
      expect(shallow(<StringHighlighter {...props} highlightClass="a-class" />)).toMatchSnapshot();
    });

    it('should render correctly when disabled', () => {
      expect(shallow(<StringHighlighter {...props} enabled={false} />)).toMatchSnapshot();
    });
  });
});
