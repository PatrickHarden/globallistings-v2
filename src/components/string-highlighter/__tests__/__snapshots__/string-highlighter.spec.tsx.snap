// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components string-highlighter should render correctly 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "This is <strong>a str</strong>ing",
    }
  }
/>
`;

exports[`components string-highlighter should render correctly when disabled 1`] = `
<span>
  This is a string
</span>
`;

exports[`components string-highlighter should render correctly when there is no match 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "This is a string",
    }
  }
/>
`;

exports[`components string-highlighter should render correctly with a custom class 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "This is <strong class=a-class>a str</strong>ing",
    }
  }
/>
`;

exports[`components string-highlighter should render correctly with a custom element 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "This is <span>a str</span>ing",
    }
  }
/>
`;
